### PR TITLE
repair debug module core dump

### PR DESCRIPTION
**Pull request by oracleloyall**
Using debug mode，using mount、umount syscall，system coredump！

### CODE REVIEW

File: src/dattobd.c [hunk 1]
```diff
@@ -4778,7 +4778,22 @@ static asmlinkage long mount_hook(char __user *dev_name, char __user *dir_name,
 	//get rid of the magic value if its present
 	if((real_flags & MS_MGC_MSK) == MS_MGC_VAL) real_flags &= ~MS_MGC_MSK;
 
-	LOG_DEBUG("detected block device mount: %s -> %s : %lu", dev_name, dir_name, real_flags);
+        /*
+	 * modify by zhaoxi copy userspace buff to kernel
+	 * date 2019/10/12
+	 */
```
Comment (crawfxrd) [inline src/dattobd.c]: This comment is not needed. Your contribution will be tracked in the git log.

File: src/dattobd.c [hunk 2]
```diff
@@ -4778,7 +4778,22 @@ static asmlinkage long mount_hook(char __user *dev_name, char __user *dir_name,
...
+	char buff_dev_name[256],buff_dir_name[256];
+	memset(buff_dev_name,0,256);
+	memset(buff_dir_name,0,256);
+	copy_from_user(buff_dev_name,dev_name,256);
+	copy_from_user(buff_dir_name,dir_name,256);
+	//LOG_DEBUG("detected block device mount: %s", buff_dev_name);
```
Comment (crawfxrd) [inline src/dattobd.c]: Remove unused log

File: src/dattobd.c [hunk 3]
```diff
@@ -4778,7 +4778,22 @@ static asmlinkage long mount_hook(char __user *dev_name, char __user *dir_name,
...
+	LOG_DEBUG("detected block device mount: %s -> %s : 0x%x", buff_dev_name,
+			buff_dir_name, real_flags);
+#if 0
```
Comment (crawfxrd) [inline src/dattobd.c]: Remove dead code block

File: src/dattobd.c [hunk 4]
```diff
@@ -4778,7 +4778,22 @@ static asmlinkage long mount_hook(char __user *dev_name, char __user *dir_name,
...
 	if((real_flags & MS_MGC_MSK) == MS_MGC_VAL) real_flags &= ~MS_MGC_MSK;
 
-	LOG_DEBUG("detected block device mount: %s -> %s : %lu", dev_name, dir_name, real_flags);
+        /*
+	 * modify by zhaoxi copy userspace buff to kernel
+	 * date 2019/10/12
+	 */
+	char buff_dev_name[256],buff_dir_name[256];
+	memset(buff_dev_name,0,256);
+	memset(buff_dir_name,0,256);
```
Comment (crawfxrd) [inline src/dattobd.c]: No need to memset the buffer.

File: src/dattobd.c [hunk 5]
```diff
@@ -4778,7 +4778,22 @@ static asmlinkage long mount_hook(char __user *dev_name, char __user *dir_name,
...
+	copy_from_user(buff_dev_name,dev_name,256);
```
Comment (crawfxrd) [inline src/dattobd.c]: Check the result for error.

Comment (dakotarwilliams) [reply src/dattobd.c]: what about our defined `copy_string_from_user`?

Comment (crawfxrd) [reply src/dattobd.c]: It would be better to avoid the strdup/free. But, is it even safe to be logging in a syscall in the first place?

**Comment by dakotarwilliams** [comment]
Closing in favor of #208

**Pull request closed without merging**